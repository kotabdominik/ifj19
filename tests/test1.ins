zadejte číslo pro porovnání s číslem 10
