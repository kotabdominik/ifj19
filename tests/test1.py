a = inputi()
if a >= 10:
  if a == 10:
    print('hodnota', a, 'se rovna 10')
  else:
    print('hodnota', a, 'je vetsi jak 10')
else:
  print('hodnota', a, 'je mensi jak 10')
