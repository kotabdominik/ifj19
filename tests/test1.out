.IFJcode19
JUMP $$MAIN
LABEL $ORD
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$POSIT
POPS LF@$POSIT
DEFVAR LF@$STRING
POPS LF@$STRING
DEFVAR LF@$CHR2INT
DEFVAR LF@$TMPLEN
STRLEN LF@$TMPLEN LF@$STRING
ADD LF@$TMPLEN LF@$TMPLEN int@-1
DEFVAR LF@$BOOLCHECK
GT LF@$BOOLCHECK LF@$POSIT LF@$TMPLEN
JUMPIFEQ $ERRORORD LF@$BOOLCHECK bool@true
LT LF@$BOOLCHECK LF@$POSIT int@0
JUMPIFEQ $ERRORORD LF@$BOOLCHECK bool@true
ADD LF@$TMPLEN LF@$TMPLEN int@1
STRI2INT LF@$RETVAL LF@$STRING LF@$POSIT
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ERRORORD
MOVE LF@$RETVAL string@None
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $BETA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL int
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $SUBSTR
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$DELKA
POPS LF@$DELKA
DEFVAR LF@$POSIT
POPS LF@$POSIT
DEFVAR LF@$STRING
POPS LF@$STRING
DEFVAR LF@$TMPLEN
STRLEN LF@$TMPLEN LF@$STRING
DEFVAR LF@$BOOLCHECK
ADD LF@$TMPLEN LF@$TMPLEN int@-1
GT LF@$BOOLCHECK LF@$POSIT LF@$TMPLEN
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
LT LF@$BOOLCHECK LF@$POSIT int@0
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
GT LF@$BOOLCHECK int@0 LF@$DELKA
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
EQ LF@$BOOLCHECK int@0 LF@$DELKA
JUMPIFEQ $ERRORSUB LF@$BOOLCHECK bool@true
ADD LF@$TMPLEN LF@$TMPLEN int@1
GETCHAR LF@$RETVAL LF@$STRING LF@$POSIT
ADD LF@$POSIT LF@$POSIT int@1
DEFVAR LF@$TMPSTR
ADD LF@$DELKA LF@$DELKA LF@$POSIT
SUB LF@$DELKA LF@$DELKA int@1
LABEL $WHILE_LOOP_B
JUMPIFEQ $WHILE_LOOP_E LF@$POSIT LF@$DELKA
JUMPIFEQ $WHILE_LOOP_E LF@$POSIT LF@$TMPLEN
GETCHAR LF@$TMPSTR LF@$STRING LF@$POSIT
CONCAT LF@$RETVAL LF@$RETVAL LF@$TMPSTR
ADD LF@$POSIT LF@$POSIT int@1
JUMP $WHILE_LOOP_B
LABEL $WHILE_LOOP_E
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ERRORSUB
MOVE LF@$RETVAL string@None
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ALFA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL string
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $GAMA
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
READ LF@$RETVAL float
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $LENGTH
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$1
POPS LF@$1
STRLEN LF@$RETVAL LF@$1
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $CHAR
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$ASC
POPS LF@$ASC
INT2CHAR LF@$RETVAL LF@$ASC
PUSHS LF@$RETVAL
POPFRAME
RETURN
LABEL $ZEROERROR
EXIT int@9
LABEL $$MAIN
CREATEFRAME
PUSHFRAME
DEFVAR GF@$VARa
CALL $BETA
POPS GF@$VARa
CREATEFRAME
PUSHFRAME
PUSHS GF@$VARa
PUSHS int@10
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL
DEFVAR LF@$1
POPS LF@$1
DEFVAR LF@$2
POPS LF@$2
LT LF@$RETVAL LF@$2 LF@$1
NOT LF@$RETVAL LF@$RETVAL
PUSHS LF@$RETVAL
POPFRAME
DEFVAR LF@$COND0x7ffff66a9c80
POPS LF@$COND0x7ffff66a9c80
JUMPIFNEQ ELSE0x7ffff66a9c80 bool@true LF@$COND0x7ffff66a9c80
CREATEFRAME
PUSHFRAME
PUSHS GF@$VARa
PUSHS int@10
EQS 
DEFVAR LF@$COND0x7ffff66aa0d0
POPS LF@$COND0x7ffff66aa0d0
JUMPIFNEQ ELSE0x7ffff66aa0d0 bool@true LF@$COND0x7ffff66aa0d0
PUSHS string@hodnota
PUSHS GF@$VARa
PUSHS string@se\032rovna\03210
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL0x7ffff66aa264
DEFVAR LF@$PEPEGA0
POPS LF@$PEPEGA0
DEFVAR LF@$PEPEGA1
POPS LF@$PEPEGA1
DEFVAR LF@$PEPEGA2
POPS LF@$PEPEGA2
WRITE LF@$PEPEGA2
WRITE string@\032
WRITE LF@$PEPEGA1
WRITE string@\032
WRITE LF@$PEPEGA0
WRITE string@\010
MOVE LF@$RETVAL0x7ffff66aa264  string@None
POPFRAME
JUMP END0x7ffff66aa0d0
LABEL ELSE0x7ffff66aa0d0
PUSHS string@hodnota
PUSHS GF@$VARa
PUSHS string@je\032vetsi\032jak\03210
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL0x7ffff66aaaf4
DEFVAR LF@$PEPEGA0
POPS LF@$PEPEGA0
DEFVAR LF@$PEPEGA1
POPS LF@$PEPEGA1
DEFVAR LF@$PEPEGA2
POPS LF@$PEPEGA2
WRITE LF@$PEPEGA2
WRITE string@\032
WRITE LF@$PEPEGA1
WRITE string@\032
WRITE LF@$PEPEGA0
WRITE string@\010
MOVE LF@$RETVAL0x7ffff66aaaf4  string@None
POPFRAME
LABEL END0x7ffff66aa0d0
POPFRAME
JUMP END0x7ffff66a9c80
LABEL ELSE0x7ffff66a9c80
PUSHS string@hodnota
PUSHS GF@$VARa
PUSHS string@je\032mensi\032jak\03210
CREATEFRAME
PUSHFRAME
DEFVAR LF@$RETVAL0x7ffff66ab0e4
DEFVAR LF@$PEPEGA0
POPS LF@$PEPEGA0
DEFVAR LF@$PEPEGA1
POPS LF@$PEPEGA1
DEFVAR LF@$PEPEGA2
POPS LF@$PEPEGA2
WRITE LF@$PEPEGA2
WRITE string@\032
WRITE LF@$PEPEGA1
WRITE string@\032
WRITE LF@$PEPEGA0
WRITE string@\010
MOVE LF@$RETVAL0x7ffff66ab0e4  string@None
POPFRAME
LABEL END0x7ffff66a9c80
POPFRAME
